src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
> Arm2 isn't a tuple struct at all, but rather, a path,

This suggests that the fix could be jsut adding `()` in the empty `pat_vec` case